specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,209 @@
+  @summary("Get list of schema groups.")
+  @doc("Gets the list of schema groups user is authorized to access.")
+  @sharedRoute
+  op listSchemaGroups is resourceOperations.ResourceList<SchemaGroup>;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/$schemas/{id}?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ SchemaId resource doesn't allow returning binary schema and headers."
+  @summary("Get a registered schema by its unique ID reference.")
+  @doc("Gets a registered schema by its unique ID.  Azure Schema Registry guarantees that ID is unique within a namespace. Operation response type is based on serialization of schema requested.")
+  @route("/$schemaGroups/$schemas/{id}")
+  @get
+  op getSchemaById is Azure.Core.Foundations.Operation<
+    Azure.Core.Foundations.ItemKeysOf<SchemaId>,
+    OkResponse & SchemaHeaders & SchemaContent & SchemaContentType
+  >;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/?api-version={apiVersion}
+  @summary("List schema versions.")
+  @doc("Gets the list of all versions of one schema.")
+  op listSchemaVersions is resourceOperations.ResourceList<Version>;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions/{schemaVersion}?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ Version resource doesn't allow returning non-version response."
+  @summary("Get specific schema versions.")
+  @doc("Gets one specific version of one schema.")
+  @route("/$schemaGroups/{groupName}/schemas/{name}/versions/{schemaVersion}")
+  @get
+  op getSchemaByVersion is Azure.Core.Foundations.Operation<
+    Azure.Core.Foundations.ItemKeysOf<Version>,
+    OkResponse & SchemaHeaders & SchemaContent & SchemaContentType
+  >;
+
+  //  POST https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}:get-id?api-version={apiVersion}
+  @summary("Get ID for existing schema.")
+  @doc("Gets the ID referencing an existing schema within the specified schema group, as matched by schema content comparison.")
+  @action("get-id")
+  op getSchemaIdByContent is resourceOperations.ResourceAction<
+    Schema,
+    {
+      ...SchemaContentType;
+
+      @doc("String representation (UTF-8) of the registered schema.")
+      @body
+      schemaContent: bytes;
+    },
+    SchemaHeaders
+  >;
+
+  //  PUT https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-names" "Register operation does not use create/replace naming"
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard createOrReplace operation does not allow passing in binary body to request."
```

Comment:
Did you open/discuss this issue with the TypeSpec team? Curious about what they said about supporting these scenarios?

Reply:
OOB discussion with Johan: I believe he's working on something that should make this scenario (passing a Resource with a body into a standard operation template) possible/easier to navigate.